src/ai/backend/manager/server.py

```diff
@@ -281,6 +292,83 @@ async def manager_status_ctx(root_ctx: RootContext) -> AsyncIterator[None]:
     yield
 
 
+@actxmgr
+async def leader_election_ctx(root_ctx: RootContext) -> AsyncIterator[None]:
+    configuration = root_ctx.local_config["manager"]["raft-configuration"]
+    raft_id = configuration[root_ctx.pidx]
+    grpc_port = int(raft_id.split(":")[-1])
+    filtered_configuration = tuple(conf for conf in configuration if conf != raft_id)
+
+    root_ctx.node_id = f"{raft_id}:{os.getpid()}"
+    reigns: bool = False
+    global_timer_ids: Final[List[str]] = []
+
+    async def _on_global_timer_created(context: None, source: str, event: GlobalTimerCreatedEvent):
+        if root_ctx.node_id == event.node_id:
+            global_timer_ids.append(event.timer_id)
+            if reigns:
+                await root_ctx.event_producer.produce_event(
+                    GlobalTimerJoinEvent(node_id=root_ctx.node_id, timer_id=event.timer_id)
+                )
+
+    global_timer_created_handler = root_ctx.event_dispatcher.subscribe(
+        GlobalTimerCreatedEvent, None, _on_global_timer_created
+    )
```

Comment:
are global timers registered by an event handler because the timers should produce event as soon as registered if the manager is leader?